### ISSUE TITLE
Error creating authorized domains with wildcards on Firefox

### ISSUE THREAD

**Issue by marandaneto** [PostHog/posthog#24453]
## Bug description

https://posthoghelp.zendesk.com/agent/tickets/17024
`Please enter a valid domain (URLs with a path aren't allowed)` error when adding a domain using Firefox

https://github.com/PostHog/posthog/blob/a2ae70be7e3a4d4dacc26cc017311613e52d3220/frontend/src/lib/utils.tsx#L599

## How to reproduce

1. Create an authorized domain using `https://*.example.com`

## Additional context

```
function isDomain(url: string): boolean {
    try {
        const parsedUrl = new URL(url)
        if (parsedUrl.protocol.includes('http') && (!parsedUrl.pathname || parsedUrl.pathname === '/')) {
            return true
        }
        if (!parsedUrl.pathname.replace(/^\/\//, '').includes('/')) {
            return true
        }
    } catch {
        return false
    }
    return false
}

console.log(isDomain('https://*.example.com'))
```

`new URL(url)` throws in Firefox since they handle `URL` different than Chrome

`new URL(url)` is used in multiple places so it's not a one-liner, most likely we need a more intelligent option using regexes only.
such as https://github.com/PostHog/posthog/blob/a2ae70be7e3a4d4dacc26cc017311613e52d3220/frontend/src/lib/components/AuthorizedUrlList/authorizedUrlListLogic.ts#L86

### Debug info

```shell
- [ ] PostHog Cloud, Debug information: [please copy/paste from https://us.posthog.com/settings/project-details#variables or https://eu.posthog.com/settings/project-details#variables]
- [ ] PostHog Hobby self-hosted with `docker compose`, version/commit: [please provide]
- [ ] PostHog self-hosted with Kubernetes (deprecated, see [`Sunsetting Kubernetes support`](https://posthog.com/blog/sunsetting-helm-support-posthog)), version/commit: [please provide]
```

**Comment by pauldambra** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/17249 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/19316)

**Comment by pauldambra** [comment]
fixed in #24583 and #24671
